tests/ui/sized/stack-overflow-trait-infer-98842.rs

```diff
@@ -0,0 +1,18 @@
+// #98842 stack overflow in trait inference
+// issue: rust-lang/rust#98842
+//@ check-fail
+//@ edition:2021
+//@ stderr-per-bitwidth
+//@ ignore-endian-big
```

Comment:
The output here doesn't look endianess-dependent, why is this ignored on big-endian targets?

Reply:
Good catch, I don't think there's a meaningful reason for that.

Reply:
Cc @uweigand might be worth testing this on a big-endian target and removing the annotation if it passes there.

(Sorry for always pinging you here, unfortunately our platform page doesn't list people to ping for our big-endian targets so I am going off of PRs like https://github.com/rust-lang/rust/pull/106047. If you are part of a group of people that are helping maintain one of our tier 2 targets, please see https://github.com/rust-lang/rust/issues/113739.)

Reply:
iirc this was to work around the output differing in https://github.com/rust-lang/rust/pull/122895#issuecomment-2015748989

Reply:
That's a bitwidth difference, but not an endianess difference.

Also we often use normalization to remove these bitwidth difference as maintaining the bessed output becomes a lot more annoying with stderr-per-bitwidth.

Reply:
> Cc @uweigand might be worth testing this on a big-endian target and removing the annotation if it passes there.
> 
> (Sorry for always pinging you here, unfortunately our platform page doesn't list people to ping for our big-endian targets so I am going off of PRs like #106047. If you are part of a group of people that are helping maintain one of our tier 2 targets, please see #113739.)

Sorry for the late reply.  I've now verified that the test continues to pass on s390x if the annotation is removed.